recipes/eigen/all/conanfile.py

```diff
@@ -0,0 +1,43 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanException
+
+class eigenConan(ConanFile):
+    name = "eigen"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://eigen.tuxfamily.org"
+    description = "Eigen is a C++ template library for linear algebra: matrices, vectors, \
+                   numerical solvers, and related algorithms."
+    license = "MPL-2.0"
+    topics = ("eigen", "algebra", "linear-algebra", "vector", "numerical")
+    _source_subfolder = "_source_subfolder"
+    no_copy_source = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        #Get te extracted folder name. They allways have the format eigen-eigen-xxxxxx
+        listdir = os.listdir()
+        extracted_dir = [i for i in listdir if "eigen-eigen" in i][0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        unsupported_folder = os.path.join(self.package_folder, "include", "eigen3", "unsupported", "Eigen")
+        eigen_folder = os.path.join(self.package_folder, "include", "eigen3", "Eigen")
+        self.copy("COPYING.*", dst="licenses", src=self._source_subfolder)
+        self.copy("*", dst=eigen_folder, src=os.path.join(self._source_subfolder, "Eigen"))
+        self.copy("*", dst=unsupported_folder, src=os.path.join(self._source_subfolder, "unsupported", "Eigen"))
+        self.copy("signature_of_eigen3_matrix_library", dst=os.path.join("include", "eigen3"), src=self._source_subfolder)
+        os.remove(os.path.join(eigen_folder, "CMakeLists.txt"))
+        os.remove(os.path.join(unsupported_folder, "CMakeLists.txt"))
+        os.remove(os.path.join(unsupported_folder, "CXX11", "CMakeLists.txt"))
+        os.remove(os.path.join(unsupported_folder, "CXX11", "src", "Tensor", "README.md"))
+        os.remove(os.path.join(unsupported_folder, "src", "EulerAngles", "CMakeLists.txt"))
+        os.rename(os.path.join(unsupported_folder, "src", "LevenbergMarquardt", "CopyrightMINPACK.txt"),
+                               os.path.join(self.package_folder, "licenses", "CopyrightMINPACK.txt"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs = [os.path.join("include","eigen3")]
```

Comment:
```suggestion
        self.cpp_info.includedirs = [os.path.join("include","eigen3")]
        if tools.os_info.is_linux:
            self.cpp_info.system_libs = ["m"]
```
https://bitbucket.org/eigen/eigen/src/94875feeeeb9abe5509b314197da1991ba2070f5/CMakeLists.txt#lines-118